kola/tests/docker/torcx_manifest_pkgs.go

```diff
@@ -0,0 +1,145 @@
+		Run:              dockerTorcxManifestPkgs,
+		ClusterSize:      0,
+		Name:             "docker.torcx-manifest-pkgs",
+		ExcludePlatforms: []string{"qemu"}, // Downloads torcx packages
+		// the first version torcx manifests were shipped
+		MinVersion: semver.Version{Major: 1535},
+	})
+}
+
+func dockerTorcxManifestPkgs(c cluster.TestCluster) {
+	if c.TorcxManifest == nil {
+		c.Skip("no torcx manifest provided")
+		return
+	}
+
+	var dockerPkgs *cluster.TorcxPackage
+	for _, pkg := range c.TorcxManifest.Packages {
+		pkg := pkg
+		if pkg.Name == "docker" {
+			dockerPkgs = &pkg
+			break
+		}
+	}
+	if dockerPkgs == nil {
+		c.Fatalf("torcx manifest provided, but didn't include docker packages: %+v", c.TorcxManifest)
+	}
+
+	// Generate an ignition config that downloads all of the docker torcx packages referenced
+	ignitionConfig := ignition.Config{
+		Ignition: ignition.Ignition{
+			Version: "2.1.0",
+		},
+		Storage: ignition.Storage{
+			Files: []ignition.File{},
+		},
+	}
+
+	for _, version := range dockerPkgs.Versions {
+		version := version
+		var url string
+		for _, loc := range version.Locations {
+			if loc.URL != nil {
+				url = *loc.URL
+			}
+		}
+		if url == "" {
+			c.Fatalf("not all docker versions had a remote location available: %+v", c.TorcxManifest)
+		}
+
+		ignitionConfig.Storage.Files = append(ignitionConfig.Storage.Files, ignition.File{
+			Node: ignition.Node{
+				Filesystem: "root",
+				Path:       fmt.Sprintf("/var/lib/torcx/store/docker:%s.torcx.tgz", version.Version),
+			},
+			FileEmbedded1: ignition.FileEmbedded1{
+				Contents: ignition.FileContents{
+					Source: url,
+					Verification: ignition.Verification{
+						Hash: &version.Hash,
+					},
+				},
+			},
+		})
+	}
+
+	ignitionBytes, err := json.Marshal(ignitionConfig)
+	if err != nil {
+		c.Fatalf("marshal err: %v", err)
+	}
+
+	m, err := c.NewMachine(conf.Ignition(string(ignitionBytes)))
+	if err != nil {
+		c.Fatalf("could not boot machine: %v", err)
+	}
+
+	// Make sure the default torcx config was fine
+	if _, err := m.SSH(`docker version`); err != nil {
+		c.Fatalf("could not run docker: %v", err)
+	}
+
+	// And now swap in a profile for each package and make sure it works
+	for _, version := range dockerPkgs.Versions {
+		version := version.Version
+		_, err := m.SSH(fmt.Sprintf(`sudo tee /etc/torcx/profiles/docker.json <<EOF
+{
+  "kind": "profile-manifest-v0",
+  "value": {
+    "images": [
+      {
+        "name": "docker",
+        "reference": "%s"
+      }
+    ]
+  }
+}
+EOF
+echo -n "docker" | sudo tee /etc/torcx/next-profile
+`, version))
+		if err != nil {
+			c.Fatalf("could not set profile: %v", err)
+		}
+
+		if err := m.Reboot(); err != nil {
+			c.Fatalf("could not reboot: %v", err)
+		}
+		currentVersion, err := m.SSH(`jq -r '.value.images[] | select(.name == "docker").reference' /run/torcx/profile.json`)
+		if err != nil {
+			c.Fatalf("could not get current docker ref: %v", err)
+		}
+		if string(currentVersion) != version {
```

Comment:
Is it feasible to cross-check this version with the one reported by docker itself?  Otherwise we're only verifying that torcx claims it did what we wanted it to do.

Reply:
We can do a `string.HasPrefix` which will be right in all current cases.

I'll add that, but since our version doesn't translate directly to docker's version, it feels a little hacky.